backend/src/services/attendances/attendances.repo.js

```diff
@@ -134,39 +148,59 @@ class AttendancesRepo {
   }
 
   /**
-   * Check if a lecture exists and belongs to the course.
+   * Check if a user is enrolled in a course.
    *
-   * @param {number} lectureId - ID of the lecture
+   * @param {number} userId - ID of the user
    * @param {number} courseId - ID of the course
-   * @returns {Promise<boolean>} True if lecture exists and belongs to course, false otherwise
+   * @returns {Promise<boolean>} True if enrolled, false otherwise
    */
-  async lectureExists(lectureId, courseId) {
-    const lecture = await this.db.lectures.findFirst({
+  async isUserEnrolledInCourse(userId, courseId) {
+    const enrollment = await this.db.enrollments.findFirst({
       where: {
-        id: lectureId,
+        user_id: userId,
         course_id: courseId,
         deleted_at: null,
       },
     });
-    return !!lecture;
+    return !!enrollment;
   }
 
   /**
-   * Check if a user is enrolled in a course.
+   * Get attendance statistics for a lecture.
+   * Returns total enrolled students, total present, and attendance percentage.
    *
-   * @param {number} userId - ID of the user
+   * @param {number} lectureId - ID of the lecture
    * @param {number} courseId - ID of the course
-   * @returns {Promise<boolean>} True if enrolled, false otherwise
+   * @returns {Promise<Object>} Statistics object with total_enrolled, total_present, attendance_percentage
    */
-  async isUserEnrolledInCourse(userId, courseId) {
-    const enrollment = await this.db.enrollments.findFirst({
+  async getAttendanceStats(lectureId, courseId) {
+    // Get total enrolled students in the course
+    const totalEnrolled = await this.db.enrollments.count({
       where: {
-        user_id: userId,
         course_id: courseId,
+        role: 'student',
```

Comment:
[Note for future]
The role can also be 'team_lead' for a student learning in the course. We can handle this in #227 .